Hyper-V: updated test cases priorities

chvalean opened the pull request:
Updated priorities for Hyper-V only test cases.
Shared Azure & Hyper-V tests have been already updated.

Separate features into individual xml files for:
- File copy (over vmbus)
- Production Checkpoint
- Runtime Memory Resize

chvalean commented on Testscripts/Linux/check_traces.sh:
```diff
@@ -0,0 +1,38 @@
+[[ -n $summary_log ]] || summary_log="/root/summary.log"
+# Checking logs
+while true; do
+	# Check for hot add errors in dmesg
```
fix indentation, same for below line 27.

asofron replied:
Fixed

chvalean commented on Testscripts/Windows/AddVhdxHardDisk.ps1:
```diff
@@ -0,0 +1,379 @@
+     The script will create a .vhdx file, and mount it to the
+     specified hard drive. If the hard drive does not exist, it
+     will be created.
+
```
remove extra blank lines

asofron replied:
removed

chvalean commented on Testscripts/Windows/AddVhdxHardDisk.ps1:
```diff
@@ -0,0 +1,379 @@
+    Test data for this test case
+
+.Example
+    setupScripts\AddVhdxHardDisk.ps1 -vmName myVM -hvServer localhost -testParams "SCSI=0,0,Dynamic,4096;sshkey=linux_id_rsa.ppk;ipv4=IPaddress;RootDir="
```
testparams must be updated

asofron replied:
updated

chvalean commented on Testscripts/Windows/FCOPY-BASIC.ps1:
```diff
@@ -0,0 +1,172 @@
+    LogMsg "Enabling  Guest services on VM:'${VMname}'"
+    Enable-VMIntegrationService -Name "Guest Service Interface" -VMname $VMname -ComputerName $HvServer
+    LogMsg "Starting VM:'${VMname}'"
+	Start-VM -Name $VMname -ComputerName $HvServer
```
fix indentation

asofron replied:
Fixed

chvalean commented on Testscripts/Windows/FCOPY-BASIC.ps1:
```diff
@@ -0,0 +1,172 @@
+if ($Error.Count -eq 0) {
+	LogMsg "File has been successfully copied to guest VM '${VMname}'" 
+}
+elseif (($Error.Count -gt 0) -and ($Error[0].Exception.Message -like "*FAIL to initiate copying files to the guest: The file exists. (0x80070050)*")) {
```
line too long

asofron replied:
Fixed

chvalean commented on Testscripts/Windows/FCOPY-DISABLE-ENABLE.ps1:
```diff
@@ -0,0 +1,262 @@
+{
+    return "Skipped"
+}
+
```
remove blank lines

asofron replied:
Fixed

chvalean commented on Testscripts/Windows/FCOPY-NON-ASCII.ps1:
```diff
@@ -0,0 +1,250 @@
+# Sending the test file to VM
+#
+$Error.Clear()
+Copy-VMFile -vmName $VMName -ComputerName $HvServer -SourcePath $filePath -DestinationPath "/tmp/" -FileSource host -ErrorAction SilentlyContinue
```
line too long, same for line 196

asofron replied:
Fixed

chvalean commented on Testscripts/Windows/RemoveVhdxHardDisk.ps1:
```diff
@@ -0,0 +1,397 @@
+   SCSI=0,0,Dynamic,4096 : Add a hard drive on SCSI controller 0, Lun 0, vhd type of Dynamic disk with logical sector size of 4096
+   IDE=1,1,Fixed,4096  : Add a hard drive on IDE controller 1, IDE port 1, vhd type of Fixed disk with logical sector size of 4096
+
+   A typical XML definition for this test case would look similar to the following:
```
remove xml definition example, no longer applicable

asofron replied:
Removed RemoveVhdxHardDisk.ps1 as Lisa handles the removing of disk so there no more need for the script

chvalean commented on Testscripts/Windows/RemoveVhdxHardDisk.ps1:
```diff
@@ -0,0 +1,397 @@
+    Test data for this test case
+
+.Example
+    setupScripts\RemoveVhdxHardDisk -vmName VM -hvServer localhost -testParams "SCSI=0,0,Dynamic,4096;sshkey=rhel5_id_rsa.ppk;ipv4=IP;RootDir="
```
testparams must be updated

asofron replied:
Removed RemoveVhdxHardDisk.ps1

chvalean commented on Testscripts/Windows/RemoveVhdxHardDisk.ps1:
```diff
@@ -0,0 +1,397 @@
+
+    switch ($fields[0].Trim())
+    {
+    "SSHKey"    { $sshKey  = $fields[1].Trim() }
```
this should no longer be used

juhlee-microsoft commented on Libraries/CommonFunctions.psm1:
```diff
@@ -1066,1551 +1066,2491 @@ Function DeployVMs ($xmlConfig, $setupType, $Distro, $getLogsIfFailed = $false,
+        }
+        else
+        {
+              
```
delete an extra line

juhlee-microsoft commented on Libraries/CommonFunctions.psm1:
```diff
@@ -1066,1551 +1066,2491 @@ Function DeployVMs ($xmlConfig, $setupType, $Distro, $getLogsIfFailed = $false,
+                {
+                    LogMsg "Unable to authenticate. Not retrying!"
+                    Throw "Calling function - $($MyInvocation.MyCommand). Unable to authenticate"
+
```
delete an extra line

chvalean commented on XML/TestCases/FunctionalTests.xml:
```diff
@@ -245,4 +245,93 @@
+        <Platform>HyperV</Platform>
+        <Category>FCOPY</Category>
+        <Area>FCOPY</Area>
+        <Tags>fcopy</Tags>
```
We introduce new tags in order to be able to target the different features in Hyper-V.
These are part of extended testing - functional, and we should not cram 200 test cases under BVT or Core.

iamshital commented on Libraries/CommonFunctions.psm1:
```diff
@@ -2875,3 +2875,1115 @@ function CheckVMState {
+       LogMsg "Systemd not found on VM"
+       $check1 = $false
+    }
+    .\Tools\plink.exe -C -pw $Password -P $SSHPort $Username@$Ipv4 "systemd-analyze --help"
```
Do we have this scenario where first check is false and second check true on any distro?

chvalean replied:
This is a double-check for systemd - first look for systemd and then for systemd-analyze.
Then only if both conditions are met we confirm that systemd is present, this function is to handle older distros that use initd and we skip the tests in such situations.

iamshital commented on Testscripts/Windows/AddVhdxHardDisk.ps1:
```diff
@@ -0,0 +1,374 @@
+#     If the -SCSI options is false, an IDE drive is created
+#
+############################################################################
+function Create-HardDrive( [string] $vmName, [string] $server, [System.Boolean] $SCSI, [int] $ControllerID,
```
This should go in HyperV.psm1

asofron replied:
This is used a setup script some of the test cases so i don`t know if HyperV.psm1 is the place for it

juhlee-microsoft replied:
@asofron I think this function is only for Hyper-V test only. Can you move to HyperV.ps1, if you agree?

chvalean replied:
The script is usually ran before the VM is started, so we cannot integrate this into other modules.
It would also imply a refactor of all storage tests to follow a new model.

asofron commented on Testscripts/Windows/FCOPY-NON-ASCII.ps1:
```diff
@@ -0,0 +1,250 @@
+#
+if ($gsi.OperationalStatus -ne "OK") {
+   LogErr "The Guest services are not working properly for VM '${vmName}'!" 
+   return = "FAIL"
```
Fixed replaced with return "FAIL"

asofron commented on Testscripts/Windows/FCOPY-NON-ASCII.ps1:
```diff
@@ -0,0 +1,250 @@
+$sts = Check-FcopyDaemon  -vmPassword $VMPassword -VmPort $VMPort -vmUserName $VMUserName -ipv4 $Ipv4
+if (-not $sts[-1]) {
+	 LogErr "File copy daemon is not running inside the Linux guest VM!" 
+	 return = "FAIL"
```
Fixed replaced with return "FAIL"

asofron commented on XML/TestCases/FunctionalTests.xml:
```diff
@@ -245,4 +245,93 @@
+        <setupType>SingleVM</setupType>
+        <Platform>HyperV</Platform>
+        <Category>FCOPY</Category>
+        <Area>FCOPY</Area>
```
Corrected to be consistent

asofron commented on Testscripts/Windows/FCOPY-REPEATED-DELETE.ps1:
```diff
@@ -0,0 +1,190 @@
+    return "Skipped"
+}
+
+$retVal = "PASS"
```
Removed positive value

asofron commented on Testscripts/Windows/FCOPY-REPEATED-DELETE.ps1:
```diff
@@ -0,0 +1,190 @@
+	# Waiting for the VM to run again and respond to SSH - port 22
+	do {
+		sleep 5
+	} until (Test-NetConnection $Ipv4 -Port 22 -WarningAction SilentlyContinue | ? { $_.TcpTestSucceeded } )
```
Updated with variable

asofron commented on Testscripts/Windows/FCOPY-OVERWRITE.ps1:
```diff
@@ -0,0 +1,163 @@
+
+
+
+$retVal = "PASS"
```
Removed $retval from testcase

asofron commented on Testscripts/Windows/FCOPY-REPEATED-DELETE.ps1:
```diff
@@ -0,0 +1,190 @@
+$gsi = Get-VMIntegrationService -vmName $VMName -ComputerName $HvServer -Name "Guest Service Interface"
+if (-not $gsi) {
+    LogErr "Error: Unable to retrieve Integration Service status from VM '${vmName}'" 
+    return "Aborted"
```
Fixed

asofron commented on Testscripts/Windows/FCOPY-REPEATED-DELETE.ps1:
```diff
@@ -0,0 +1,190 @@
+$BuildNumber = Get-HostBuildNumber  -hvServer $HvServer
+if ($BuildNumber -eq 0)
+{
+    return "FAIL"
```
Removed $retVal and used return instead

asofron commented on Testscripts/Windows/FCOPY-OVERWRITE.ps1:
```diff
@@ -0,0 +1,163 @@
+$gsi = Get-VMIntegrationService -vmName $VMName -ComputerName $HvServer -Name "Guest Service Interface"
+if (-not $gsi) {
+    LogErr "Error: Unable to retrieve Integration Service status from VM '${vmName}'" 
+    return "Aborted"
```
Updated

asofron commented on Testscripts/Windows/FCOPY-OVERWRITE.ps1:
```diff
@@ -0,0 +1,163 @@
+}
+elseif ($BuildNumber -lt 9600)
+{
+    return $Skipped
```
replaced with "ABORTED"

asofron commented on Testscripts/Windows/FCOPY-NON-ASCII.ps1:
```diff
@@ -0,0 +1,250 @@
+}
+elseif ($BuildNumber -lt 9600)
+{
+    return "Skipped"
```
replaced with "ABORTED"

asofron commented on Testscripts/Windows/FCOPY-NEGATIVE.ps1:
```diff
@@ -0,0 +1,202 @@
+
+
+$retVal = "FAIL"
+$testfile = $null
```
updated

asofron commented on Testscripts/Windows/FCOPY-LARGE-FILE.ps1:
```diff
@@ -0,0 +1,187 @@
+
+if ($gsi.OperationalStatus -ne "OK") {
+	LogErr "The Guest services are not working properly for VM '${vmName}'!" 
+	$retVal = "FAIL"
```
Updated

asofron commented on Testscripts/Windows/FCOPY-NEGATIVE.ps1:
```diff
@@ -0,0 +1,202 @@
+}
+elseif ($BuildNumber -lt 9600){
+    LogMsg "Hyper-v host version $BuildNumber does not support fcopy, skipping test." 
+    return "Skipped"
```
Updated

asofron commented on Testscripts/Windows/FCOPY-LARGE-FILE.ps1:
```diff
@@ -0,0 +1,187 @@
+$gsi = Get-VMIntegrationService -vmName $VMName -ComputerName $HvServer -Name "Guest Service Interface"
+if (-not $gsi) {
+    LogErr "Unable to retrieve Integration Service status from VM '${vmName}'" 
+    return "Aborted"
```
Upedated

asofron commented on Testscripts/Windows/FCOPY-LARGE-FILE.ps1:
```diff
@@ -0,0 +1,187 @@
+}
+elseif ($BuildNumber -lt 9600)
+{
+    return "Skipped"
```
Updated

asofron commented on Testscripts/Windows/FCOPY-LARGE-FILE.ps1:
```diff
@@ -0,0 +1,187 @@
+$sts = Check-FcopyDaemon  -vmPassword $VMPassword -VmPort $VMPort -vmUserName $VMUserName -ipv4 $Ipv4
+if (-not $sts[-1]) {
+	 LogErr "File copy daemon is not running inside the Linux guest VM!" 
+	$retVal = "FAIL"
```
Updated

asofron commented on Testscripts/Windows/FCOPY-REPEATED-DELETE.ps1:
```diff
@@ -0,0 +1,190 @@
+
+if ($gsi.OperationalStatus -ne "OK") {
+   LogErr "The Guest services are not working properly for VM '${vmName}'!" 
+    $retVal = "FAIL"
```
Removed  '$retVal = "FAIL" and used return 'FAIL'

asofron commented on Testscripts/Windows/FCOPY-LARGE-FILE.ps1:
```diff
@@ -0,0 +1,187 @@
+    LogERR "ERROR: Cannot remove the test file '${testfile}'!" 
+}
+
+return $retVal
```
Updated

asofron commented on Testscripts/Windows/FCOPY-LARGE-FILE.ps1:
```diff
@@ -0,0 +1,187 @@
+{
+    return "Skipped"
+}
+$retVal = "PASS"
```
Omited $retVal

asofron commented on Testscripts/Windows/FCOPY-LARGE-FILE.ps1:
```diff
@@ -0,0 +1,187 @@
+$testfile = $null
+$gsi = $null
+# Default 10GB file size
+$filesize = 10737418240
```
In case no parameters are provided by XML a default size of 10GB is used

asofron commented on Testscripts/Windows/FCOPY-FILE-EXISTS.ps1:
```diff
@@ -0,0 +1,189 @@
+$gsi = Get-VMIntegrationService -vmName $VMName -ComputerName $hvServer -Name "Guest Service Interface"
+if (-not $gsi) {
+    LogErr " Unable to retrieve Integration Service status from VM '${vmName}'" 
+    return "Aborted"
```
Updated

asofron commented on Testscripts/Windows/FCOPY-FILE-EXISTS.ps1:
```diff
@@ -0,0 +1,189 @@
+    return "Abort"
+}
+
+$retVal = "PASS"
```
Omitted

asofron commented on Testscripts/Windows/FCOPY-FILE-EXISTS.ps1:
```diff
@@ -0,0 +1,189 @@
+}
+
+
+return $retVal
```
removed return $retVal

asofron commented on Testscripts/Windows/FCOPY-FILE-EXISTS.ps1:
```diff
@@ -0,0 +1,189 @@
+        $TestParams
+    )
+
+$retVal = "FAIL"
```
$retVal = "FAIL" omitted

asofron commented on Testscripts/Windows/FCOPY-DISABLE-ENABLE.ps1:
```diff
@@ -0,0 +1,262 @@
+    $hvPathFormatted = $hvPath.Replace(':','$')
+
+    # Define the file-name to use with the current time-stamp
+    $testfile = "testfile-$(get-date -uformat '%H-%M-%S-%Y-%m-%d').file"
```
Updated

asofron commented on Testscripts/Windows/FCOPY-DISABLE-ENABLE.ps1:
```diff
@@ -0,0 +1,262 @@
+}
+elseif ($BuildNumber -lt 9600)
+{
+    return "Skipped"
```
updated

asofron commented on Testscripts/Windows/FCOPY-BASIC.ps1:
```diff
@@ -0,0 +1,172 @@
+    LogErr "cannot remove the test file '${testfile}'!" 
+}
+
+return $retVal
```
removed return $retVal

asofron commented on Testscripts/Windows/FCOPY-BASIC.ps1:
```diff
@@ -0,0 +1,172 @@
+$gsi = Get-VMIntegrationService -VMname $VMname -ComputerName $HvServer -Name "Guest Service Interface"
+if (-not $gsi) {
+    LogErr "Unable to retrieve Integration Service status from VM '${VMname}'" 
+    return "Aborted"
```
updated

asofron commented on Testscripts/Windows/FCOPY-BASIC.ps1:
```diff
@@ -0,0 +1,172 @@
+}
+
+
+$retVal = "PASS"
```
Omitted

asofron commented on Testscripts/Windows/FCOPY-BASIC.ps1:
```diff
@@ -0,0 +1,172 @@
+}
+elseif ($BuildNumber -lt 9600)
+{
+    return "Aborted"
```
Updated

asofron commented on Libraries/CommonFunctions.psm1:
```diff
@@ -2875,3 +2875,1115 @@ function CheckVMState {
+
+    .\Tools\plink.exe -C -pw $vmPassword -P $vmPort $vmUserName@$ipv4 "ps -ef | grep '[h]v_fcopy_daemon\|[h]ypervfcopyd' > /tmp/fcopy_present"
+    if (-not $?) {
+        LogErr  "ERROR: Unable to verify if the fcopy daemon is running" 
```
Update and removed extra info

asofron commented on Testscripts/Windows/FCOPY-BASIC.ps1:
```diff
@@ -0,0 +1,172 @@
+        $TestParams
+    )
+
+$retVal = "FAIL"
```
Omitted

asofron commented on Libraries/CommonFunctions.psm1:
```diff
@@ -2875,3 +2875,1115 @@ function CheckVMState {
+
+    $filename = ".\fcopy_present"
+
+    .\Tools\plink.exe -C -pw $vmPassword -P $vmPort $vmUserName@$ipv4 "ps -ef | grep '[h]v_fcopy_daemon\|[h]ypervfcopyd' > /tmp/fcopy_present"
```
Send-CommandToVM is used by some test case in this PR , also it used by functions within CommonFunctions.psm1

iamshital replied:
But still I see that plink.exe is being used directly at lot of places. 
Is there any difference, the way plink is interactiting with guest VMs? or is it some HyperV Guest related prerequisite/limitation (that I'm not aware of) that we can not use a common method to send command to Linux Guest?

asofron commented on Testscripts/Windows/FCOPY-DISABLE-ENABLE.ps1:
```diff
@@ -0,0 +1,262 @@
+#
+#######################################################################
+# Read parameters
+$params = $testParams.TrimEnd(";").Split(";")
```
Updated

chvalean commented:
@juhlee-microsoft @iamshital please check the test results posted in the initial PR comment and do a final code check, this should be ready to merge.

iamshital commented on Testscripts/Windows/FCOPY-LARGE-FILE.ps1:
```diff
@@ -0,0 +1,176 @@
+    LogErr "ERROR: Cannot remove the test file '${testfile}'!" 
+    return "FAIL"
+}
+}
```
Looks like wrong indentation here.

iamshital commented on Testscripts/Windows/FCOPY-NEGATIVE.ps1:
```diff
@@ -0,0 +1,194 @@
+return "PASS"
+}
+
+}
```
Wrong indentation.

iamshital commented on Testscripts/Windows/FCOPY-NON-ASCII.ps1:
```diff
@@ -0,0 +1,243 @@
+#
+LogMsg "Test completed successfully"
+return "PASS"
+}
```
Wrong indentation.

iamshital commented on Testscripts/Windows/FCOPY-OVERWRITE.ps1:
```diff
@@ -0,0 +1,149 @@
+    LogErr "Cannot remove the test file '${testfile}'!" 
+    return "FAIL"
+}
+}
```
Wrong indentation.

iamshital commented on Testscripts/Windows/FCOPY-REPEATED-DELETE.ps1:
```diff
@@ -0,0 +1,165 @@
+    LogErr "Cannot remove the test file '${testfile}'!" 
+    return "FAIL"
+}
+}
```
Wrong indentation.

iamshital commented on Libraries/CommonFunctions.psm1:
```diff
@@ -2875,3 +2875,1115 @@ function CheckVMState {
 
     return $vmStatus
 }
+function Check-File{
```
Please give easily identifiable name to function. Ex. Check-FileInLinuxGuest.

iamshital commented on Libraries/CommonFunctions.psm1:
```diff
@@ -2875,3 +2875,1115 @@ function CheckVMState {
+    return $True
+}
+
+function Remove-FileVM{
```
Correct me if I'm wrong, but this function only executes one line of code.
`Send-CommandToVM -vmPassword $vmPassword -vmPort $vmPort -ipv4 $ipv4 "rm -f /mnt/$testfile"`.
Can't this command be used directly?

asofron replied:
Removed function and replace in with Send-CommandToVM -vmPassword $vmPassword -vmPort $vmPort -ipv4 $ipv4 "rm -f /mnt/$testfile"

iamshital commented on Libraries/CommonFunctions.psm1:
```diff
@@ -2875,3 +2875,1115 @@ function CheckVMState {
+        if set checkContent as $True, will return file content.
+   #>
+    if ($checkSize) {
+        
```
Extra line is okay if there is a logical code separation.

iamshital commented on Libraries/CommonFunctions.psm1:
```diff
@@ -2875,3 +2875,1115 @@ function CheckVMState {
+        return $False
+    }
+    if ($checkContent) {
+        
```
Extra line is okay if there is a logical code separation.

iamshital replied:
If possible, can you please auto format this PR? For bash and powershell.

juhlee-microsoft replied:
Please run PowerShell Script Analyzer tool.

chvalean committed:
Hyper-V: updated test cases priorities

Updated priorities for Hyper-V only test cases.
Shared Azure & Hyper-V tests have been already updated.

Separate features into individual xml files for:
- File copy (over vmbus)
- Production Checkpoint
- Runtime Memory Resize

juhlee-microsoft reviewed (approved):
Please change the priority of Net-Max-NICs
